lib/service/businessLogic.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const kinveyCompletionHandler = require('./kinveyCompletionHandler');
+const kinveyErrors = require('kinvey-datalink-errors');
+const notImplementedHandler = require('./notImplementedHandler');
+
+const registeredFunctions = new Map();
+
+function register(taskName, functionToExecute) {
```

Comment:
@mjsalinger Data links have a set of methods for every serviceObject, hence the `serviceObject` class, whereas BL simply has a set of handlers which are arbitrarily created and mapped to hooks, and said mappings are stored in MAPI BL metadata (e.g. `onPreSaveServer`). Backend-sdk simply needs to store the hooks by name in a single map and the metadata determines when they are called